psutil/mem.go

```diff
@@ -26,75 +26,82 @@ import (
 	"github.com/shirou/gopsutil/mem"
 )
 
-func virtualMemory(ns core.Namespace) (*plugin.MetricType, error) {
+func virtualMemory(nss []core.Namespace) ([]plugin.MetricType, error) {
 	mem, err := mem.VirtualMemory()
 	if err != nil {
 		return nil, err
 	}
 
-	switch ns.String() {
-	case "/intel/psutil/vm/total":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Total,
-			Unit_:      "B",
-		}, nil
-	case "/intel/psutil/vm/available":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Available,
-			Unit_:      "B",
-		}, nil
-	case "/intel/psutil/vm/used":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Used,
-			Unit_:      "B",
-		}, nil
-	case "/intel/psutil/vm/used_percent":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.UsedPercent,
-		}, nil
-	case "/intel/psutil/vm/free":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Free,
-			Unit_:      "B",
-		}, nil
-	case "/intel/psutil/vm/active":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Active,
-			Unit_:      "B",
-		}, nil
-	case "/intel/psutil/vm/inactive":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Inactive,
-			Unit_:      "B",
-		}, nil
-	case "/intel/psutil/vm/buffers":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Buffers,
-			Unit_:      "B",
-		}, nil
-	case "/intel/psutil/vm/cached":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Cached,
-			Unit_:      "B",
-		}, nil
-	case "/intel/psutil/vm/wired":
-		return &plugin.MetricType{
-			Namespace_: ns,
-			Data_:      mem.Wired,
-			Unit_:      "B",
-		}, nil
+	results := make([]plugin.MetricType, len(nss))
+
+	for i, ns := range nss {
+
```

Comment:
It doesn't change performances, but with the same idea to make the code clear, it might be more readable to set `Data_` and `Unit_` in variables before creating the metric:
```
		var data interface{}
		unit := "B"

		switch ns.Element(len(ns) - 1).Value {
 		case "total":
 			data = mem.Total
		// ...
 		case "used_percent":
 			data = mem.UsedPercent
 			unit = ""
		// ...
		default:
			return nil, fmt.Errorf("Requested memory statistic %s is not found", ns.String())
		}

		results[i] = plugin.MetricType{
			Namespace_: ns,
			Data_:     data,
			Unit_:      unit,
		}
```